docs/references/execution-errors.mdx

```diff
@@ -386,14 +386,346 @@ There is a [limit][limits] on the total size in bytes of the [code section](http
+The request to create a canister did not include enough cycles to pay the creation fee.
+
+An example of this error is:
+```
+  Creating a canister requires a fee of 100000000000 that is deducted from the canister's initial balance but only 1000000 cycles were received with the create_canister request.
+```
+
+To fix this error, include enough cycles with the request to meet the creation fee.
+
+
+### Install code not enough cycles
+A canister doesn't have enough cycles to execute an install message.
+
+An example of this error is:
+```
+  Canister installation failed with `Canister xxx-xxx is out of cycles: please top up the canister with at least 10000000000 additional cycles`.
+```
+
+To fix this error, consider [topping up][topping-up] the canister with additional cycles.
+
+
+### Install code rate limited
+An install message for the given canister cannot be executed because the canister has been rate limited.
+
+An example of this error is:
+```
+  Canister xxx-xxx is rate limited because it executed too many instructions in the previous install_code messages. Please retry installation after several minutes.
+```
+
+There is a limit on how many instructions a canister can execute for installations in a given time period. To fix this error, consider retrying the installation at a later time.
+
+
+### Subnet out of canister ids
+There was an attempt to create a canister on a subnet which cannot hold any more canisters.
+
+An example of this error is:
+```
+  Could not create canister. Subnet has surpassed its canister ID allocation.
+```
+
+To fix this error, consider creating the canister on another subnet.
+
+
+### Invalid settings
+A canister was created with invalid settings or there was an attempt to change the canister settings to an invalid value.
+
+An example of this error is:
+```
+  Could not validate the settings: Invalid settings: 'controllers' length exceeds maximum size allowed of 10.
+```
+
+To fix this error, apply the described changes to make the settings valid.
+
+
+### Maximum number of canisters reached
+There was an attempt to create a canister on a subnet which has reached the maximum number of allowed canisters.
+
+An example of this error is:
+```
+  Subnet yyy-yyy has reached the allowed canister limit of 100000 canisters. Retry creating the canister.
+```
+
+To fix this error, consider creating the canister on another subnet.
+
+
+### Insufficient cycles in compute allocation
+The canister does not have enough cycles to increase its compute allocation.
+
+An example of this error is:
+```
+  Cannot increase compute allocation to 50 due to insufficient cycles. At least 1000000000 additional cycles are required.
+```
+
+Canisters need to pay for [compute allocation][compute-allocation] and there was an attempt to increase the compute allocation of a canister beyond what it can currently pay for. To fix this error, consider [adding the required cycles][topping-up] to the canister.
```

Comment:
```suggestion
Canisters need to pay for [compute allocation][compute-allocation] and there was an attempt to increase the compute allocation of a canister beyond what it can currently pay for. To fix this error, consider [topping up][topping-up] to the canister.
```